prime-router/src/main/kotlin/history/azure/SubmissionFunction.kt

```diff
@@ -126,4 +128,31 @@ class SubmissionFunction(
     ): HttpResponseMessage {
         return this.getDetailedView(request, id)
     }
+
+    @FunctionName("getTiMetadataForHistory")
+    fun getTiMetadata(
+        @HttpTrigger(
+            name = "getTiMetadataForHistory",
+            methods = [HttpMethod.GET],
+            authLevel = AuthorizationLevel.ANONYMOUS,
+            route = "waters/report/{id}/history/etorMetadata"
+        ) request: HttpRequestMessage<String?>,
+        @BindingName("id") id: String,
+        context: ExecutionContext,
+    ): HttpResponseMessage {
+        return this.retrieveMetadata(request, id, context, null)
+    }
+
+    override fun getLookupId(id: String): String {
+        val submissionActionId = this.actionFromId(id)
+        val submissionHistory = submissionsFacade.findDetailedSubmissionHistory(submissionActionId)
```

Comment:
It seems like it would be cleaner to use logic similar to findDetailedDeliveryHistory here, where we simply, in this case, walk DOWN the tree (instead of up like in findDetailedDeliveryHistory). ReportGraph.kt should ultimately be used by both of your endpoints because it is more straight forward.

So, I am suggesting a new method in ReportGraph called something like "getDescendentReports" and "getDescendentReport" that uses `reportDescendantGraphCommonTableExpression`

Also, in findDetailedDeliveryHistory, why not just pass in the reportID?

It seems the logic in the change set related to getLookupId could be greatly simplified if the above is implemented.

Reply:
Just to check, after we create a new method in ReportGraph, are you suggesting having SubmissionFunction call ReportGraph directly? Other methods in SubmissionFunction call SubmissionFacade which calls ReportService which calls ReportGraph